scipy/stats/stats.py

```diff
@@ -2089,29 +2102,99 @@ def percentileofscore(a, score, kind='rank'):
     >>> stats.percentileofscore([1, 2, 3, 3, 4], 3, kind='mean')
     60.0
 
+    Multidimensional score arrays are supported:
+
+    >>> stats.percentileofscore([1, 2, 3, 3, 4], [2, 3])
+    array([40., 70.])
+
+    The inputs can be infinite:
+
+    >>> stats.percentileofscore([-np.inf, 0, 1, np.inf], [1, 100])
+    array([75., 75.])
+
+    If `a` is empty, or containing only nan's (which are omitted),
+    the resulting percentiles are all nan:
+
+    >>> stats.percentileofscore([], 1)
+    nan
+    >>> stats.percentileofscore([np.nan, 1, 2], [1, 2], nan_policy='omit')
+    array([ 50., 100.])
+    >>> stats.percentileofscore([np.nan, np.nan], [1, 2], nan_policy='omit')
+    array([nan, nan])
+
+    The 'omit' policy does not make sense if there are nan's among the scores,
+    and so an error in that case.
     """
-    if np.isnan(score):
-        return np.nan
+
     a = np.asarray(a)
     n = len(a)
+    score = np.asarray(score)
+
+    # Nan treatment
+    cna, npa = _contains_nan(a, nan_policy, summary_check=False)
+    cns, nps = _contains_nan(score, nan_policy, summary_check=False)
+
+    if (cna or cns) and nan_policy == 'raise':
+        raise ValueError("The input contains nan values")
+
+    elif cns and nan_policy == "omit":
+        raise ValueError("The input scores contains nan values,"
+                         " which is incompatible with the 'omit' policy.")
```

Comment:
I don't think this is consistent with "[A Design Specification for `nan_policy`](https://docs.scipy.org/doc/scipy/reference/dev/api-dev/nan_policy.html)".
![image](https://user-images.githubusercontent.com/6570539/115665630-6715c080-a2f8-11eb-8d77-841da326c45f.png)

Is there a reason that NaNs in `score` cannot be removed?

As a minor note, that document consistently uses "`nan`" and "`nan` values". [`np.isnan` documentation](https://numpy.org/doc/stable/reference/generated/numpy.isnan.html) refers to them as "NaN". I know SciPy stats documentation is not consistent about it, but while we're here, I'd suggest "`nan`s" or "NaNs" instead of "nan's".

Reply:
> Is there a reason that NaNs in score cannot be removed?

Yes, consider

```
    score = [
        [a, b],
        [c, nan],
    ]

    This should yield:
    result = [
        [pa, pb],
        [pc, ???],
    ]
```

Clearly, the `nan` cannot be omitted. The design specification for `nan_policy` does not seem to consider such cases.

Reply:
I see. @tupui you already approved, but what do you think about this?

Reply:
To me it did not make much sense to have nan in the scores as well, but I never work with these so I am not sure. That's why I asked your help here 😅

Reply:
Yeah... I really don't like `nan_policy = 'omit'`. That's the motivation behind gh-13312; so we fix it once and for all for most functions so we can forget about it. (But that won't work here, so yes, this function does need special consideration.) 

I'm really not sure what to do, but I think when in doubt about interface decisions like this, the mailing list is good thing to try.

Reply:
I think gh-14221 says this would be fine as-is. I'd suggest commenting on that issue @patricknraanes to indicate your opinion between the two options. Once we resolve that, i think this is close.

Reply:
It looks like #14221 will break in favour of "option 1". I therefore made the changes to support it.

Reply:
Any movement on this?

Reply:
I agree that issue seems to have converged to option 1 -- so one of the `stats` folks just needs to give this a final "ok" or are more changes needed to fully support option 1?